go/arrow/array/string.go

```diff
@@ -60,7 +60,7 @@ func (a *String) String() string {
 		case a.IsNull(i):
 			o.WriteString("(null)")
 		default:
-			fmt.Fprintf(o, "%q", a.Value(i))
+			fmt.Fprintf(o, "%q", a.Value(i+a.array.data.offset))
```

Comment:
hum... this doesn't sound right :)
I don't think that's the right place to fix that.
I believe it should be either in `Value`, in `NewSliceData` or in `MakeFromData`.

(thanks for doing that!)

Reply:
good point Value might be better suited